upgrade QGIS to 3.10.2

Both, [nixos-19.09](https://nixos.org/nixos/packages.html?attr=qgis&channel=nixos-19.09&query=qgis) and [nixos-unstable currently ship QGIS 3.10.**1**](https://nixos.org/nixos/packages.html?attr=qgis&channel=nixos-unstable&query=qgis).

In their blog Post ["Public Service Announcement: Update to the latest point release now"](http://blog.qgis.org/2020/01/24/public-service-announcement-update-to-the-latest-point-release-now/), the QGIS core developers recommend all QGIS 3.10.x users to update to QGIS 3.10.**2** as soon as possible:

> QGIS users who have adopted the 3.10 version when initially released at the end of October 2019 have likely noticed a sharp drop in reliability. The underlying issues have now been addressed in 3.10.2, all users are advised to update *now*.

Some of the fixes were in QGIS itself, some in the underlying libraries GDAL/OGR and PROJ. So these should be upgraded as well:

> When QGIS 3.10 was first released in the end of October 2019, a pair of libraries – namely GDAL and PROJ – were updated to their next-generation versions. [...]
>
> The update of these crucial libraries led to a number of regressions. While we expected some issues to arise, the seriousness of the disruption caught us off guard. Yet, it was also somewhat inevitable: QGIS is the first large GIS project to expose these next-generation libraries to the masses. The large number of QGIS users across the globe were essentially stress testing both new code within QGIS as well as the libraries themselves.

("next-generation versions" probably refers to a change in the major versions.)

The versions that should be used are

* GDAL/OGR 3.0.3
* PROJ 6.3.0

as I've figured out in [my [QGIS-Developer] mailing list post here](https://lists.osgeo.org/pipermail/qgis-developer/2020-January/060056.html).

Both nixos-19.09 and nixos-unstable currently ship PROJ 6.1.1 and GDAL/OGR 3.0.1.

Thus, I see the following things to do:

- [x] `gdal`: 3.0.1 -> 3.0.3 on `master` (PR #78774)
- [ ] `proj`: 6.1.1 -> 6.3.0 on `master` (PR #83317)
- [ ] `qgis`: 3.10.1 -> 3.10.2 on `master` (PR #83317)

- [x] `gdal`: 3.0.1 -> 3.0.3 on `release-19.09` (PR #79257)
- [ ] `proj`: 6.1.1 -> 6.3.0 on `release-19.09`
- [ ] `qgis`: 3.10.1 -> 3.10.2 on `release-19.09`

/cc qgis maintainer @lsix
/cc gdal maintainer @MarcWeber
/cc proj maintainer @vbgl

pyproj 2.5.0 with support for PROJ 6.3 (pyproj4/pyproj#487) has been released:

- pyproj4/pyproj#547
- https://github.com/pyproj4/pyproj/releases/tag/v2.5.0rel

However, I can't seem to get it to build with `nix-build -I nixpkgs=. -A python3Packages.pyproj`:

If I bump only the pyproj version to 2.5.0 and leave PROJ at 6.1.1, I get an import error when the build attempts pyproj's tests (they don't even seem to run). This is probably expected, as [the pyproj documentation](https://pyproj4.github.io/pyproj/v2.5.0rel/) lists PROJ 6.2.0 as the minimum supported version.

```
running install tests
ImportError while loading conftest '/build/source/test/conftest.py'.
test/conftest.py:7: in <module>
    import pyproj
/nix/store/ql3ygc3v1sbxkqlcg2b32n99zqfax6z8-python3.7-pyproj-2.5.0/lib/python3.7/site-packages/pyproj/__init__.py:70: in <module>
    from pyproj import _datadir
E   ImportError: /nix/store/ql3ygc3v1sbxkqlcg2b32n99zqfax6z8-python3.7-pyproj-2.5.0/lib/python3.7/site-packages/pyproj/_datadir.cpython-37m-x86_64-linux-gnu.so: undefined symbol: proj_context_set_autoclose_database
builder for '/nix/store/a45ibdz14lfpzdi15hp5zcrnvdrnaf3y-python3.7-pyproj-2.5.0.drv' failed with exit code 4
error: build of '/nix/store/a45ibdz14lfpzdi15hp5zcrnvdrnaf3y-python3.7-pyproj-2.5.0.drv' failed
```

When I bump pyproj to 2.5.0 **and** PROJ to 6.2.1, then the tests _do_ run, but 23 fail and one errors.

<details><summary>test output (click to expand)</summary>

```
running install tests
============================= test session starts ==============================
platform linux -- Python 3.7.5, pytest-5.1.0, py-1.8.0, pluggy-0.12.0
rootdir: /build/source
collected 497 items / 10 deselected / 487 selected                             

test/test__main__.py ...                                                 [  0%]
test/test_awips221.py .                                                  [  0%]
test/test_datum.py ..                                                    [  1%]
test/test_datum_shift.py ....                                            [  2%]
test/test_exception_logging.py ..                                        [  2%]
test/test_geod.py .......FFFFFFFFFFFFFFFFFFFFF                           [  8%]
test/test_list.py ...................                                    [ 12%]
test/test_pickle.py .                                                    [ 12%]
test/test_proj.py ........................                               [ 17%]
test/test_show_versions.py ....                                          [ 18%]
test/test_transform.py ..                                                [ 18%]
test/test_transformer.py ............................................... [ 28%]
..FF......                                                               [ 30%]
test/crs/test_crs.py ................................................... [ 40%]
........................................................................ [ 55%]
...............................................................          [ 68%]
test/crs/test_crs_cf.py ......................s...........               [ 75%]
test/crs/test_crs_coordinate_operation.py .............................. [ 81%]
.................                                                        [ 85%]
test/crs/test_crs_coordinate_system.py ......................            [ 89%]
test/crs/test_crs_datum.py .......                                       [ 90%]
test/crs/test_crs_json.py ...............................                [ 97%]
test/crs/test_crs_maker.py .............E                                [100%]

==================================== ERRORS ====================================
_________________ ERROR at teardown of test_bound_crs__example _________________

    @pytest.fixture(scope="session")
    def aoi_data_directory():
        """
        This is to ensure that the ntv2_0.gsb file is actually
        missing for the AOI tests.
        """
        data_dir = pyproj.datadir.get_data_dir()
        with tempfile.TemporaryDirectory() as tmpdir:
            tmp_data_dir = os.path.join(tmpdir, "proj")
            shutil.copytree(data_dir, tmp_data_dir)
            try:
                os.remove(os.path.join(str(tmp_data_dir), "ntv2_0.gsb"))
            except OSError:
                pass
            try:
                pyproj.datadir.set_data_dir(str(tmp_data_dir))
                yield
            finally:
>               pyproj.datadir.set_data_dir(data_dir)

test/conftest.py:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/tempfile.py:805: in __exit__
    self.cleanup()
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/tempfile.py:809: in cleanup
    _shutil.rmtree(self.name)
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:494: in rmtree
    _rmtree_safe_fd(fd, path, onerror)
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:432: in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:452: in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

topfd = 22, path = '/build/tmphc137m_b/proj'
onerror = <function rmtree.<locals>.onerror at 0x7fffe2df2830>

    def _rmtree_safe_fd(topfd, path, onerror):
        try:
            with os.scandir(topfd) as scandir_it:
                entries = list(scandir_it)
        except OSError as err:
            err.filename = path
            onerror(os.scandir, path, sys.exc_info())
            return
        for entry in entries:
            fullname = os.path.join(path, entry.name)
            try:
                is_dir = entry.is_dir(follow_symlinks=False)
                if is_dir:
                    orig_st = entry.stat(follow_symlinks=False)
                    is_dir = stat.S_ISDIR(orig_st.st_mode)
            except OSError:
                is_dir = False
            if is_dir:
                try:
                    dirfd = os.open(entry.name, os.O_RDONLY, dir_fd=topfd)
                except OSError:
                    onerror(os.open, fullname, sys.exc_info())
                else:
                    try:
                        if os.path.samestat(orig_st, os.fstat(dirfd)):
                            _rmtree_safe_fd(dirfd, fullname, onerror)
                            try:
                                os.rmdir(entry.name, dir_fd=topfd)
                            except OSError:
                                onerror(os.rmdir, fullname, sys.exc_info())
                        else:
                            try:
                                # This can only happen if someone replaces
                                # a directory with a symlink after the call to
                                # os.scandir or stat.S_ISDIR above.
                                raise OSError("Cannot call rmtree on a symbolic "
                                              "link")
                            except OSError:
                                onerror(os.path.islink, fullname, sys.exc_info())
                    finally:
                        os.close(dirfd)
            else:
                try:
>                   os.unlink(entry.name, dir_fd=topfd)
E                   PermissionError: [Errno 13] Permission denied: 'GL27'

/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:450: PermissionError
=================================== FAILURES ===================================
_________________________ test_geometry_length__point __________________________

    @skip_shapely
    def test_geometry_length__point():
        geod = Geod(ellps="WGS84")
>       assert geod.geometry_length(Point(1, 2)) == 0
E       NameError: name 'Point' is not defined

test/test_geod.py:185: NameError
_______________________ test_geometry_length__linestring _______________________

    @skip_shapely
    def test_geometry_length__linestring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
>           geod.geometry_length(LineString([Point(1, 2), Point(3, 4)])),
            313588.39721259556,
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:192: NameError
__________________ test_geometry_length__linestring__radians ___________________

    @skip_shapely
    def test_geometry_length__linestring__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               LineString(
                    [
                        Point(math.radians(1), math.radians(2)),
                        Point(math.radians(3), math.radians(4)),
                    ]
                ),
                radians=True,
            ),
            313588.39721259556,
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:203: NameError
_______________________ test_geometry_length__linearring _______________________

    @skip_shapely
    def test_geometry_length__linearring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               LinearRing(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            1072185.2103813463,
            decimal=2,
        )
E       NameError: name 'LinearRing' is not defined

test/test_geod.py:221: NameError
________________________ test_geometry_length__polygon _________________________

    @skip_shapely
    def test_geometry_length__polygon():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            1072185.2103813463,
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:233: NameError
____________________ test_geometry_length__polygon__radians ____________________

    @skip_shapely
    def test_geometry_length__polygon__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               Polygon(
                    LineString(
                        [
                            Point(math.radians(1), math.radians(2)),
                            Point(math.radians(3), math.radians(4)),
                            Point(math.radians(5), math.radians(2)),
                        ]
                    )
                ),
                radians=True,
            ),
            1072185.2103813463,
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:245: NameError
______________________ test_geometry_length__multipolygon ______________________

    @skip_shapely
    def test_geometry_length__multipolygon():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
E       NameError: name 'Polygon' is not defined

test/test_geod.py:264: NameError
_________________ test_geometry_length__multipolygon__radians __________________

    @skip_shapely
    def test_geometry_length__multipolygon__radians():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(
            LineString(
                [
                    Point(math.radians(1), math.radians(2)),
                    Point(math.radians(3), math.radians(4)),
                    Point(math.radians(5), math.radians(2)),
                ]
            )
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:275: NameError
____________________ test_geometry_length__multilinestring _____________________

    @skip_shapely
    def test_geometry_length__multilinestring():
        geod = Geod(ellps="WGS84")
>       line_string = LineString([Point(1, 2), Point(3, 4), Point(5, 2)])
E       NameError: name 'LineString' is not defined

test/test_geod.py:294: NameError
_______________________ test_geometry_length__multipoint _______________________

    @skip_shapely
    def test_geometry_length__multipoint():
        geod = Geod(ellps="WGS84")
>       assert (
            geod.geometry_length(MultiPoint([Point(1, 2), Point(3, 4), Point(5, 2)])) == 0
        )
E       NameError: name 'Point' is not defined

test/test_geod.py:305: NameError
_____________________ test_geometry_area_perimeter__point ______________________

    @skip_shapely
    def test_geometry_area_perimeter__point():
        geod = Geod(ellps="WGS84")
>       assert geod.geometry_area_perimeter(Point(1, 2)) == (0, 0)
E       NameError: name 'Point' is not defined

test/test_geod.py:313: NameError
___________________ test_geometry_area_perimeter__linestring ___________________

    @skip_shapely
    def test_geometry_area_perimeter__linestring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
>           geod.geometry_area_perimeter(LineString([Point(1, 2), Point(3, 4)])),
            (0.0, 627176.7944251911),
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:320: NameError
______________ test_geometry_area_perimeter__linestring__radians _______________

    @skip_shapely
    def test_geometry_area_perimeter__linestring__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               LineString(
                    [
                        Point(math.radians(1), math.radians(2)),
                        Point(math.radians(3), math.radians(4)),
                    ]
                ),
                radians=True,
            ),
            (0.0, 627176.7944251911),
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:331: NameError
___________________ test_geometry_area_perimeter__linearring ___________________

    @skip_shapely
    def test_geometry_area_perimeter__linearring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               LinearRing(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            (-49187690467.58623, 1072185.2103813463),
            decimal=2,
        )
E       NameError: name 'LinearRing' is not defined

test/test_geod.py:349: NameError
____________________ test_geometry_area_perimeter__polygon _____________________

    @skip_shapely
    def test_geometry_area_perimeter__polygon():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            (-49187690467.58623, 1072185.2103813463),
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:361: NameError
________________ test_geometry_area_perimeter__polygon__radians ________________

    @skip_shapely
    def test_geometry_area_perimeter__polygon__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               Polygon(
                    LineString(
                        [
                            Point(math.radians(1), math.radians(2)),
                            Point(math.radians(3), math.radians(4)),
                            Point(math.radians(5), math.radians(2)),
                        ]
                    )
                ),
                radians=True,
            ),
            (-49187690467.58623, 1072185.2103813463),
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:373: NameError
_________________ test_geometry_area_perimeter__polygon__holes _________________

    @skip_shapely
    def test_geometry_area_perimeter__polygon__holes():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               Polygon(
                    LineString([Point(1, 1), Point(1, 10), Point(10, 10), Point(10, 1)]),
                    holes=[LineString([Point(1, 2), Point(3, 4), Point(5, 2)])],
                )
            ),
            (-944373881400.3394, 3979008.0359657984),
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:394: NameError
__________________ test_geometry_area_perimeter__multipolygon __________________

    @skip_shapely
    def test_geometry_area_perimeter__multipolygon():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
E       NameError: name 'Polygon' is not defined

test/test_geod.py:407: NameError
_____________ test_geometry_area_perimeter__multipolygon__radians ______________

    @skip_shapely
    def test_geometry_area_perimeter__multipolygon__radians():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(
            LineString(
                [
                    Point(math.radians(1), math.radians(2)),
                    Point(math.radians(3), math.radians(4)),
                    Point(math.radians(5), math.radians(2)),
                ]
            )
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:418: NameError
________________ test_geometry_area_perimeter__multilinestring _________________

    @skip_shapely
    def test_geometry_area_perimeter__multilinestring():
        geod = Geod(ellps="WGS84")
>       line_string = LineString([Point(1, 2), Point(3, 4), Point(5, 2)])
E       NameError: name 'LineString' is not defined

test/test_geod.py:437: NameError
___________________ test_geometry_area_perimeter__multipoint ___________________

    @skip_shapely
    def test_geometry_area_perimeter__multipoint():
        geod = Geod(ellps="WGS84")
>       assert geod.geometry_area_perimeter(
            MultiPoint([Point(1, 2), Point(3, 4), Point(5, 2)])
        ) == (0, 0)
E       NameError: name 'Point' is not defined

test/test_geod.py:448: NameError
_____________________ test_transformer_group__unavailable ______________________

aoi_data_directory = None

    def test_transformer_group__unavailable(aoi_data_directory):
        trans_group = TransformerGroup(4326, 2964)
>       assert len(trans_group.unavailable_operations) == 1
E       assert 2 == 1
E        +  where 2 = len([<Coordinate Operation: Concatenated Operation>\nInverse of NAD27 to WGS 84 (85) + Alaska Albers\nArea of Use:\n- name: U... of NAD27 to WGS 84 (33) + Alaska Albers\nArea of Use:\n- name: Canada - NAD27\n- bounds: (-141.01, 40.04, -47.74, 83.17)])
E        +    where [<Coordinate Operation: Concatenated Operation>\nInverse of NAD27 to WGS 84 (85) + Alaska Albers\nArea of Use:\n- name: U... of NAD27 to WGS 84 (33) + Alaska Albers\nArea of Use:\n- name: Canada - NAD27\n- bounds: (-141.01, 40.04, -47.74, 83.17)] = <TransformerGroup: best_available=False>\n- transformers: 7\n- unavailable_operations: 2.unavailable_operations

test/test_transformer.py:517: AssertionError
______________________ test_transform_group__missing_best ______________________

aoi_data_directory = None

    def test_transform_group__missing_best(aoi_data_directory):
        with pytest.warns(FutureWarning):
            lat_lon_proj = pyproj.Proj(init="epsg:4326", preserve_units=False)
            alaska_aea_proj = pyproj.Proj(init="epsg:2964", preserve_units=False)
    
        with pytest.warns(
            UserWarning, match="Best transformation is not available due to missing Grid"
        ):
            trans_group = pyproj.transformer.TransformerGroup(
                lat_lon_proj.crs, alaska_aea_proj.crs
            )
    
        assert not trans_group.best_available
>       assert len(trans_group.transformers) == 37
E       assert 30 == 37
E        +  where 30 = len([<Concatenated Operation Transformer: pipeline>\nDescription: axis order change (2D) + Inverse of NAD27 to WGS 84 (3) +... axis order change (2D) + unknown\nArea of Use:\n- name: Mexico - onshore\n- bounds: (-118.47, 14.51, -86.68, 32.72), ...])
E        +    where [<Concatenated Operation Transformer: pipeline>\nDescription: axis order change (2D) + Inverse of NAD27 to WGS 84 (3) +... axis order change (2D) + unknown\nArea of Use:\n- name: Mexico - onshore\n- bounds: (-118.47, 14.51, -86.68, 32.72), ...] = <TransformerGroup: best_available=False>\n- transformers: 30\n- unavailable_operations: 48.transformers

test/test_transformer.py:539: AssertionError
=============================== warnings summary ===============================
test/test_transformer.py::test_transformer_group__unavailable
  /nix/store/y4aljzfwlqqgq7hb2jwjy4rcjyfp05al-python3.7-pyproj-2.5.0/lib/python3.7/site-packages/pyproj/transformer.py:101: UserWarning: Best transformation is not available due to missing Grid(short_name=alaska, full_name=, package_name=proj-datumgrid, url=https://download.osgeo.org/proj/proj-datumgrid-latest.zip, direct_download=True, open_license=True, available=False)
    area_of_interest=area_of_interest,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
= 23 failed, 463 passed, 1 skipped, 10 deselected, 1 warnings, 1 error in 12.96s =
builder for '/nix/store/j21z00ay1fvp23i983l19y1lpagf4zig-python3.7-pyproj-2.5.0.drv' failed with exit code 1
error: build of '/nix/store/j21z00ay1fvp23i983l19y1lpagf4zig-python3.7-pyproj-2.5.0.drv' failed
```

</details>

When I bump pyproj to 2.5.0 **and** PROJ to 6.3.0 or 6.3.1, then the tests _do_ run, but 25 fail and one errors.

<details><summary>test output (click to expand)</summary>

```
running install tests
============================= test session starts ==============================
platform linux -- Python 3.7.5, pytest-5.1.0, py-1.8.0, pluggy-0.12.0
rootdir: /build/source
collected 497 items / 10 deselected / 487 selected                             

test/test__main__.py ...                                                 [  0%]
test/test_awips221.py .                                                  [  0%]
test/test_datum.py FF                                                    [  1%]
test/test_datum_shift.py ....                                            [  2%]
test/test_exception_logging.py ..                                        [  2%]
test/test_geod.py .......FFFFFFFFFFFFFFFFFFFFF                           [  8%]
test/test_list.py ...................                                    [ 12%]
test/test_pickle.py .                                                    [ 12%]
test/test_proj.py ........................                               [ 17%]
test/test_show_versions.py ....                                          [ 18%]
test/test_transform.py ..                                                [ 18%]
test/test_transformer.py ............................................... [ 28%]
..FF......                                                               [ 30%]
test/crs/test_crs.py ................................................... [ 40%]
........................................................................ [ 55%]
...............................................................          [ 68%]
test/crs/test_crs_cf.py ..................................               [ 75%]
test/crs/test_crs_coordinate_operation.py .............................. [ 81%]
.................                                                        [ 85%]
test/crs/test_crs_coordinate_system.py ......................            [ 89%]
test/crs/test_crs_datum.py .......                                       [ 90%]
test/crs/test_crs_json.py ...............................                [ 97%]
test/crs/test_crs_maker.py .............E                                [100%]

==================================== ERRORS ====================================
_________________ ERROR at teardown of test_bound_crs__example _________________

    @pytest.fixture(scope="session")
    def aoi_data_directory():
        """
        This is to ensure that the ntv2_0.gsb file is actually
        missing for the AOI tests.
        """
        data_dir = pyproj.datadir.get_data_dir()
        with tempfile.TemporaryDirectory() as tmpdir:
            tmp_data_dir = os.path.join(tmpdir, "proj")
            shutil.copytree(data_dir, tmp_data_dir)
            try:
                os.remove(os.path.join(str(tmp_data_dir), "ntv2_0.gsb"))
            except OSError:
                pass
            try:
                pyproj.datadir.set_data_dir(str(tmp_data_dir))
                yield
            finally:
>               pyproj.datadir.set_data_dir(data_dir)

test/conftest.py:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/tempfile.py:805: in __exit__
    self.cleanup()
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/tempfile.py:809: in cleanup
    _shutil.rmtree(self.name)
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:494: in rmtree
    _rmtree_safe_fd(fd, path, onerror)
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:432: in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:452: in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

topfd = 22, path = '/build/tmp1zyus2_x/proj'
onerror = <function rmtree.<locals>.onerror at 0x7fffe2997950>

    def _rmtree_safe_fd(topfd, path, onerror):
        try:
            with os.scandir(topfd) as scandir_it:
                entries = list(scandir_it)
        except OSError as err:
            err.filename = path
            onerror(os.scandir, path, sys.exc_info())
            return
        for entry in entries:
            fullname = os.path.join(path, entry.name)
            try:
                is_dir = entry.is_dir(follow_symlinks=False)
                if is_dir:
                    orig_st = entry.stat(follow_symlinks=False)
                    is_dir = stat.S_ISDIR(orig_st.st_mode)
            except OSError:
                is_dir = False
            if is_dir:
                try:
                    dirfd = os.open(entry.name, os.O_RDONLY, dir_fd=topfd)
                except OSError:
                    onerror(os.open, fullname, sys.exc_info())
                else:
                    try:
                        if os.path.samestat(orig_st, os.fstat(dirfd)):
                            _rmtree_safe_fd(dirfd, fullname, onerror)
                            try:
                                os.rmdir(entry.name, dir_fd=topfd)
                            except OSError:
                                onerror(os.rmdir, fullname, sys.exc_info())
                        else:
                            try:
                                # This can only happen if someone replaces
                                # a directory with a symlink after the call to
                                # os.scandir or stat.S_ISDIR above.
                                raise OSError("Cannot call rmtree on a symbolic "
                                              "link")
                            except OSError:
                                onerror(os.path.islink, fullname, sys.exc_info())
                    finally:
                        os.close(dirfd)
            else:
                try:
>                   os.unlink(entry.name, dir_fd=topfd)
E                   PermissionError: [Errno 13] Permission denied: 'GL27'

/nix/store/k5rdcbcwwpvj7l9f1yvd5mfggcfz16kk-python3-3.7.5/lib/python3.7/shutil.py:450: PermissionError
=================================== FAILURES ===================================
_______________________________ test_datum[Proj] _______________________________

proj_class = <class 'pyproj.proj.Proj'>, aoi_data_directory = None

    @pytest.mark.parametrize("proj_class", [Proj, CRS])
    def test_datum(proj_class, aoi_data_directory):
        p1 = proj_class(proj="latlong", datum="WGS84")
        s_1 = -111.5
        s_2 = 45.25919444444
        p2 = proj_class(proj="utm", zone=10, datum="NAD27")
        x2, y2 = transform(p1, p2, s_1, s_2)
        if LooseVersion(proj_version_str) < LooseVersion("6.3.0"):
            assert_almost_equal((x2, y2), (1402291.0833290431, 5076289.591846835))
        else:
            # https://github.com/OSGeo/PROJ/issues/1808
>           assert_almost_equal((x2, y2), (1402285.9829252, 5076292.4212746))
E           AssertionError: 
E           Arrays are not almost equal to 7 decimals
E           
E           Mismatch: 100%
E           Max absolute difference: 4.21450818
E           Max relative difference: 1.82410064e-06
E            x: array([1402288.540836 , 5076296.6357828])
E            y: array([1402285.9829252, 5076292.4212746])

test/test_datum.py:20: AssertionError
_______________________________ test_datum[CRS] ________________________________

proj_class = <class 'pyproj.crs.crs.CRS'>, aoi_data_directory = None

    @pytest.mark.parametrize("proj_class", [Proj, CRS])
    def test_datum(proj_class, aoi_data_directory):
        p1 = proj_class(proj="latlong", datum="WGS84")
        s_1 = -111.5
        s_2 = 45.25919444444
        p2 = proj_class(proj="utm", zone=10, datum="NAD27")
        x2, y2 = transform(p1, p2, s_1, s_2)
        if LooseVersion(proj_version_str) < LooseVersion("6.3.0"):
            assert_almost_equal((x2, y2), (1402291.0833290431, 5076289.591846835))
        else:
            # https://github.com/OSGeo/PROJ/issues/1808
>           assert_almost_equal((x2, y2), (1402285.9829252, 5076292.4212746))
E           AssertionError: 
E           Arrays are not almost equal to 7 decimals
E           
E           Mismatch: 100%
E           Max absolute difference: 4.21450818
E           Max relative difference: 1.82410064e-06
E            x: array([1402288.540836 , 5076296.6357828])
E            y: array([1402285.9829252, 5076292.4212746])

test/test_datum.py:20: AssertionError
_________________________ test_geometry_length__point __________________________

    @skip_shapely
    def test_geometry_length__point():
        geod = Geod(ellps="WGS84")
>       assert geod.geometry_length(Point(1, 2)) == 0
E       NameError: name 'Point' is not defined

test/test_geod.py:185: NameError
_______________________ test_geometry_length__linestring _______________________

    @skip_shapely
    def test_geometry_length__linestring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
>           geod.geometry_length(LineString([Point(1, 2), Point(3, 4)])),
            313588.39721259556,
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:192: NameError
__________________ test_geometry_length__linestring__radians ___________________

    @skip_shapely
    def test_geometry_length__linestring__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               LineString(
                    [
                        Point(math.radians(1), math.radians(2)),
                        Point(math.radians(3), math.radians(4)),
                    ]
                ),
                radians=True,
            ),
            313588.39721259556,
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:203: NameError
_______________________ test_geometry_length__linearring _______________________

    @skip_shapely
    def test_geometry_length__linearring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               LinearRing(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            1072185.2103813463,
            decimal=2,
        )
E       NameError: name 'LinearRing' is not defined

test/test_geod.py:221: NameError
________________________ test_geometry_length__polygon _________________________

    @skip_shapely
    def test_geometry_length__polygon():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            1072185.2103813463,
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:233: NameError
____________________ test_geometry_length__polygon__radians ____________________

    @skip_shapely
    def test_geometry_length__polygon__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_length(
>               Polygon(
                    LineString(
                        [
                            Point(math.radians(1), math.radians(2)),
                            Point(math.radians(3), math.radians(4)),
                            Point(math.radians(5), math.radians(2)),
                        ]
                    )
                ),
                radians=True,
            ),
            1072185.2103813463,
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:245: NameError
______________________ test_geometry_length__multipolygon ______________________

    @skip_shapely
    def test_geometry_length__multipolygon():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
E       NameError: name 'Polygon' is not defined

test/test_geod.py:264: NameError
_________________ test_geometry_length__multipolygon__radians __________________

    @skip_shapely
    def test_geometry_length__multipolygon__radians():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(
            LineString(
                [
                    Point(math.radians(1), math.radians(2)),
                    Point(math.radians(3), math.radians(4)),
                    Point(math.radians(5), math.radians(2)),
                ]
            )
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:275: NameError
____________________ test_geometry_length__multilinestring _____________________

    @skip_shapely
    def test_geometry_length__multilinestring():
        geod = Geod(ellps="WGS84")
>       line_string = LineString([Point(1, 2), Point(3, 4), Point(5, 2)])
E       NameError: name 'LineString' is not defined

test/test_geod.py:294: NameError
_______________________ test_geometry_length__multipoint _______________________

    @skip_shapely
    def test_geometry_length__multipoint():
        geod = Geod(ellps="WGS84")
>       assert (
            geod.geometry_length(MultiPoint([Point(1, 2), Point(3, 4), Point(5, 2)])) == 0
        )
E       NameError: name 'Point' is not defined

test/test_geod.py:305: NameError
_____________________ test_geometry_area_perimeter__point ______________________

    @skip_shapely
    def test_geometry_area_perimeter__point():
        geod = Geod(ellps="WGS84")
>       assert geod.geometry_area_perimeter(Point(1, 2)) == (0, 0)
E       NameError: name 'Point' is not defined

test/test_geod.py:313: NameError
___________________ test_geometry_area_perimeter__linestring ___________________

    @skip_shapely
    def test_geometry_area_perimeter__linestring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
>           geod.geometry_area_perimeter(LineString([Point(1, 2), Point(3, 4)])),
            (0.0, 627176.7944251911),
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:320: NameError
______________ test_geometry_area_perimeter__linestring__radians _______________

    @skip_shapely
    def test_geometry_area_perimeter__linestring__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               LineString(
                    [
                        Point(math.radians(1), math.radians(2)),
                        Point(math.radians(3), math.radians(4)),
                    ]
                ),
                radians=True,
            ),
            (0.0, 627176.7944251911),
            decimal=2,
        )
E       NameError: name 'LineString' is not defined

test/test_geod.py:331: NameError
___________________ test_geometry_area_perimeter__linearring ___________________

    @skip_shapely
    def test_geometry_area_perimeter__linearring():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               LinearRing(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            (-49187690467.58623, 1072185.2103813463),
            decimal=2,
        )
E       NameError: name 'LinearRing' is not defined

test/test_geod.py:349: NameError
____________________ test_geometry_area_perimeter__polygon _____________________

    @skip_shapely
    def test_geometry_area_perimeter__polygon():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
            ),
            (-49187690467.58623, 1072185.2103813463),
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:361: NameError
________________ test_geometry_area_perimeter__polygon__radians ________________

    @skip_shapely
    def test_geometry_area_perimeter__polygon__radians():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               Polygon(
                    LineString(
                        [
                            Point(math.radians(1), math.radians(2)),
                            Point(math.radians(3), math.radians(4)),
                            Point(math.radians(5), math.radians(2)),
                        ]
                    )
                ),
                radians=True,
            ),
            (-49187690467.58623, 1072185.2103813463),
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:373: NameError
_________________ test_geometry_area_perimeter__polygon__holes _________________

    @skip_shapely
    def test_geometry_area_perimeter__polygon__holes():
        geod = Geod(ellps="WGS84")
        assert_almost_equal(
            geod.geometry_area_perimeter(
>               Polygon(
                    LineString([Point(1, 1), Point(1, 10), Point(10, 10), Point(10, 1)]),
                    holes=[LineString([Point(1, 2), Point(3, 4), Point(5, 2)])],
                )
            ),
            (-944373881400.3394, 3979008.0359657984),
            decimal=2,
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:394: NameError
__________________ test_geometry_area_perimeter__multipolygon __________________

    @skip_shapely
    def test_geometry_area_perimeter__multipolygon():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(LineString([Point(1, 2), Point(3, 4), Point(5, 2)]))
E       NameError: name 'Polygon' is not defined

test/test_geod.py:407: NameError
_____________ test_geometry_area_perimeter__multipolygon__radians ______________

    @skip_shapely
    def test_geometry_area_perimeter__multipolygon__radians():
        geod = Geod(ellps="WGS84")
>       polygon = Polygon(
            LineString(
                [
                    Point(math.radians(1), math.radians(2)),
                    Point(math.radians(3), math.radians(4)),
                    Point(math.radians(5), math.radians(2)),
                ]
            )
        )
E       NameError: name 'Polygon' is not defined

test/test_geod.py:418: NameError
________________ test_geometry_area_perimeter__multilinestring _________________

    @skip_shapely
    def test_geometry_area_perimeter__multilinestring():
        geod = Geod(ellps="WGS84")
>       line_string = LineString([Point(1, 2), Point(3, 4), Point(5, 2)])
E       NameError: name 'LineString' is not defined

test/test_geod.py:437: NameError
___________________ test_geometry_area_perimeter__multipoint ___________________

    @skip_shapely
    def test_geometry_area_perimeter__multipoint():
        geod = Geod(ellps="WGS84")
>       assert geod.geometry_area_perimeter(
            MultiPoint([Point(1, 2), Point(3, 4), Point(5, 2)])
        ) == (0, 0)
E       NameError: name 'Point' is not defined

test/test_geod.py:448: NameError
_____________________ test_transformer_group__unavailable ______________________

aoi_data_directory = None

    def test_transformer_group__unavailable(aoi_data_directory):
        trans_group = TransformerGroup(4326, 2964)
>       assert len(trans_group.unavailable_operations) == 1
E       assert 2 == 1
E        +  where 2 = len([<Coordinate Operation: Concatenated Operation>\nInverse of NAD27 to WGS 84 (85) + Alaska Albers\nArea of Use:\n- name: U... of NAD27 to WGS 84 (33) + Alaska Albers\nArea of Use:\n- name: Canada - NAD27\n- bounds: (-141.01, 40.04, -47.74, 83.17)])
E        +    where [<Coordinate Operation: Concatenated Operation>\nInverse of NAD27 to WGS 84 (85) + Alaska Albers\nArea of Use:\n- name: U... of NAD27 to WGS 84 (33) + Alaska Albers\nArea of Use:\n- name: Canada - NAD27\n- bounds: (-141.01, 40.04, -47.74, 83.17)] = <TransformerGroup: best_available=False>\n- transformers: 7\n- unavailable_operations: 2.unavailable_operations

test/test_transformer.py:517: AssertionError
______________________ test_transform_group__missing_best ______________________

aoi_data_directory = None

    def test_transform_group__missing_best(aoi_data_directory):
        with pytest.warns(FutureWarning):
            lat_lon_proj = pyproj.Proj(init="epsg:4326", preserve_units=False)
            alaska_aea_proj = pyproj.Proj(init="epsg:2964", preserve_units=False)
    
        with pytest.warns(
            UserWarning, match="Best transformation is not available due to missing Grid"
        ):
            trans_group = pyproj.transformer.TransformerGroup(
                lat_lon_proj.crs, alaska_aea_proj.crs
            )
    
        assert not trans_group.best_available
>       assert len(trans_group.transformers) == 37
E       assert 30 == 37
E        +  where 30 = len([<Concatenated Operation Transformer: pipeline>\nDescription: axis order change (2D) + Inverse of NAD27 to WGS 84 (3) +... axis order change (2D) + unknown\nArea of Use:\n- name: Mexico - onshore\n- bounds: (-118.47, 14.51, -86.68, 32.72), ...])
E        +    where [<Concatenated Operation Transformer: pipeline>\nDescription: axis order change (2D) + Inverse of NAD27 to WGS 84 (3) +... axis order change (2D) + unknown\nArea of Use:\n- name: Mexico - onshore\n- bounds: (-118.47, 14.51, -86.68, 32.72), ...] = <TransformerGroup: best_available=False>\n- transformers: 30\n- unavailable_operations: 48.transformers

test/test_transformer.py:539: AssertionError
=============================== warnings summary ===============================
test/test_transformer.py::test_transformer_group__unavailable
  /nix/store/9fq3p3ycappp4dw2is7p7q37dx9vz35a-python3.7-pyproj-2.5.0/lib/python3.7/site-packages/pyproj/transformer.py:101: UserWarning: Best transformation is not available due to missing Grid(short_name=alaska, full_name=, package_name=proj-datumgrid, url=https://download.osgeo.org/proj/proj-datumgrid-1.8.zip, direct_download=True, open_license=True, available=False)
    area_of_interest=area_of_interest,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===== 25 failed, 462 passed, 10 deselected, 1 warnings, 1 error in 13.50s ======
builder for '/nix/store/xk59fjmf1zkcwid2s10bd5vi54bl1az6-python3.7-pyproj-2.5.0.drv' failed with exit code 1
error: build of '/nix/store/xk59fjmf1zkcwid2s10bd5vi54bl1az6-python3.7-pyproj-2.5.0.drv' failed
```

</details>

Are these test failures NixPkgs-specific or should I ask at the pyproj project about them?

--- a/pkgs/applications/gis/qgis/unwrapped.nix
+++ b/pkgs/applications/gis/qgis/unwrapped.nix
@@ -10,7 +10,7 @@ let
     [ qscintilla-qt5 gdal jinja2 numpy psycopg2
       chardet dateutil pyyaml pytz requests urllib3 pygments pyqt5 sip owslib six ];
 in mkDerivation rec {
-  version = "3.10.1";
+  version = "3.10.4";
   pname = "qgis";
   name = "${pname}-unwrapped-${version}";
 
@@ -18,7 +18,7 @@ in mkDerivation rec {
     owner = "qgis";
     repo = "QGIS";
     rev = "final-${lib.replaceStrings ["."] ["_"] version}";
-    sha256 = "0xq0nnp7zdqaihqvh5rsi1129g23vnk5ijkpxfzaggafgmhf5hgz";
+    sha256 = "0d1rsgjgnnq6jgms5bgppz8lkh4518nf90fk0qvxajdfi9j4jn12";
   };
 
   passthru = {

--- a/pkgs/development/libraries/proj/default.nix
+++ b/pkgs/development/libraries/proj/default.nix
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   name = "proj";
-  version = "6.1.1";
+  version = "6.3.1";
 
   src = fetchFromGitHub {
     owner = "OSGeo";
     repo = "PROJ";
     rev = version;
-    sha256 = "0w2v2l22kv0xzq5hwl7n8ki6an8vfsr0lg0cdbkwcl4xv889ysma";
+    sha256 = "1ildcp57qsa01kvv2qxd05nqw5mg0wfkksiv9l138dbhp0s7rkxp";
   };
 
   outputs = [ "out" "dev"];

--- a/pkgs/development/python-modules/pyproj/001.proj.patch
+++ b/pkgs/development/python-modules/pyproj/001.proj.patch
@@ -1,47 +1,62 @@
-diff a/pyproj/datadir.py b/pyproj/datadir.py
---- a/pyproj/datadir.py
-+++ b/pyproj/datadir.py
-@@ -52,6 +52,7 @@ def get_data_dir():
-     str: The valid data directory.
-
-     """
-+    return "@proj@/share/proj"
-     # to avoid re-validating
-     global _VALIDATED_PROJ_DATA
+diff -Nur a/pyproj/datadir.py b/pyproj/datadir.py
+--- a/pyproj/datadir.py	2020-03-24 12:53:39.417440608 +0100
++++ b/pyproj/datadir.py	2020-03-24 12:56:19.870089479 +0100
+@@ -66,9 +66,7 @@
      if _VALIDATED_PROJ_DATA is not None:
-diff a/setup.py b/setup.py
---- a/setup.py
-+++ b/setup.py
-@@ -16,7 +16,7 @@ INTERNAL_PROJ_DIR = os.path.join(CURRENT_FILE_PATH, "pyproj", BASE_INTERNAL_PROJ
-
+         return _VALIDATED_PROJ_DATA
+     global _USER_PROJ_DATA
+-    internal_datadir = os.path.join(
+-        os.path.dirname(os.path.abspath(__file__)), "proj_dir", "share", "proj"
+-    )
++    internal_datadir = "@proj@/share/proj"
+     proj_lib_dirs = os.environ.get("PROJ_LIB", "")
+     prefix_datadir = os.path.join(sys.prefix, "share", "proj")
+ 
+diff -Nur a/setup.py b/setup.py
+--- a/setup.py	2020-03-24 12:53:39.415440624 +0100
++++ b/setup.py	2020-03-24 12:52:05.311232522 +0100
+@@ -11,7 +11,7 @@
+ PROJ_MIN_VERSION = parse_version("6.2.0")
+ CURRENT_FILE_PATH = os.path.dirname(os.path.abspath(__file__))
+ BASE_INTERNAL_PROJ_DIR = "proj_dir"
+-INTERNAL_PROJ_DIR = os.path.join(CURRENT_FILE_PATH, "pyproj", BASE_INTERNAL_PROJ_DIR)
++INTERNAL_PROJ_DIR = "@proj@"
+ 
+ 
  def check_proj_version(proj_dir):
-     """checks that the PROJ library meets the minimum version"""
--    proj = os.path.join(proj_dir, "bin", "proj")
-+    proj = "@proj@/bin/proj"
-     proj_ver_bytes = subprocess.check_output(proj, stderr=subprocess.STDOUT)
-     proj_ver_bytes = (proj_ver_bytes.decode("ascii").split()[1]).strip(",")
-     proj_version = parse_version(proj_ver_bytes)
-@@ -33,6 +33,7 @@ def get_proj_dir():
-     """
-     This function finds the base PROJ directory.
-     """
-+    return "@proj@"
-     proj_dir = os.environ.get("PROJ_DIR")
-     if proj_dir is None and os.path.exists(INTERNAL_PROJ_DIR):
-         proj_dir = INTERNAL_PROJ_DIR
-@@ -56,6 +57,7 @@ def get_proj_libdirs(proj_dir):
-     """
-     This function finds the library directories
-     """
-+    return ["@proj@/lib"]
-     proj_libdir = os.environ.get("PROJ_LIBDIR")
-     libdirs = []
-     if proj_libdir is None:
-@@ -77,6 +79,7 @@ def get_proj_incdirs(proj_dir):
-     """
-     This function finds the include directories
-     """
-+    return ["@proj@/include"]
-     proj_incdir = os.environ.get("PROJ_INCDIR")
-     incdirs = []
-     if proj_incdir is None:
+@@ -146,7 +146,7 @@
+     # By default we'll try to get options PROJ_DIR or the local version of proj
+     proj_dir = get_proj_dir()
+     library_dirs = get_proj_libdirs(proj_dir)
+-    include_dirs = get_proj_incdirs(proj_dir)
++    include_dirs = get_proj_incdirs("@projdev@")
+ 
+     # setup extension options
+     ext_options = {
+diff -Nur a/test/conftest.py b/test/conftest.py
+--- a/test/conftest.py	2020-03-24 12:53:39.417440608 +0100
++++ b/test/conftest.py	2020-03-24 23:16:47.373972786 +0100
+@@ -1,6 +1,7 @@
+ import os
+ import shutil
+ import tempfile
++import stat
+ 
+ import pytest
+ 
+@@ -17,6 +18,15 @@
+     with tempfile.TemporaryDirectory() as tmpdir:
+         tmp_data_dir = os.path.join(tmpdir, "proj")
+         shutil.copytree(data_dir, tmp_data_dir)
++
++        # Data copied from the nix store is readonly (causes cleanup problem).
++        # Make it writable.
++        for r, d, files in os.walk(tmp_data_dir):
++            os.chmod(r, os.stat(r).st_mode | stat.S_IWUSR)
++            for f in files:
++                fpath = os.path.join(r, f)
++                os.chmod(fpath, os.stat(fpath).st_mode | stat.S_IWUSR)
++
+         try:
+             os.remove(os.path.join(str(tmp_data_dir), "ntv2_0.gsb"))
+         except OSError:

--- a/pkgs/development/python-modules/pyproj/default.nix
+++ b/pkgs/development/python-modules/pyproj/default.nix
@@ -1,20 +1,22 @@
-{ lib, buildPythonPackage, fetchFromGitHub, python, pkgs, pythonOlder, substituteAll
+{ lib, buildPythonPackage, fetchFromGitHub, python, pkgs, pythonOlder, isPy27, substituteAll
 , aenum
 , cython
 , pytest
 , mock
 , numpy
+, shapely
 }:
 
 buildPythonPackage rec {
   pname = "pyproj";
-  version = "2.2.2";
+  version = "2.6.0";
+  disabled = isPy27;
 
   src = fetchFromGitHub {
     owner = "pyproj4";
     repo = "pyproj";
     rev = "v${version}rel";
-    sha256 = "0mb0jczgqh3sma69k7237i38h09gxgmvmddls9hpw4f3131f5ax7";
+    sha256 = "0fyggkbr3kp8mlq4c0r8sl5ah58bdg2mj4kzql9p3qyrkcdlgixh";
   };
 
   # force pyproj to use ${pkgs.proj}
@@ -22,13 +24,14 @@ buildPythonPackage rec {
     (substituteAll {
       src = ./001.proj.patch;
       proj = pkgs.proj;
+      projdev = pkgs.proj.dev;
     })
   ];
 
   buildInputs = [ cython pkgs.proj ];
 
   propagatedBuildInputs = [
-    numpy
+    numpy shapely
   ] ++ lib.optional (pythonOlder "3.6") aenum;
 
   checkInputs = [ pytest mock ];
@@ -38,6 +41,9 @@ buildPythonPackage rec {
   checkPhase = ''
     pytest . -k 'not alternative_grid_name \
                  and not transform_wgs84_to_alaska \
+                 and not transformer_group__unavailable \
+                 and not transform_group__missing_best \
+                 and not datum \
                  and not repr' \
             --ignore=test/test_doctest_wrapper.py \
             --ignore=test/test_datadir.py

